src/vm/wren_vm.c

```diff
@@ -1070,7 +1083,11 @@ static bool runInterpreter(WrenVM* vm)
       if (fiber->numFrames == 0)
       {
         // If this is the main fiber, we're done.
-        if (fiber->caller == NULL) return true;
+        if (fiber->caller == NULL)
+        {
+          interpretResult.status = WREN_RESULT_SUCCESS;
```

Comment:
Yeah, that's not going to work for two reasons:

1. The result may not actually be a string. It could be any type. `AS_CSTRING` doesn't do any conversion, it just interprets the object's bits as that type.

2. Even in cases where the result *is* a string, you can't expose the raw memory for it outside of the VM. The GC may collect it at any time.